src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -252,14 +252,17 @@ template <typename InterfaceType> void RadioSpinel<InterfaceType>::ResetRcp(bool
     mIsReady    = false;
     mWaitingKey = SPINEL_PROP_LAST_STATUS;
 
-    if (aResetRadio && (SendReset(SPINEL_RESET_STACK) == OT_ERROR_NONE) && (WaitResponse(false) == OT_ERROR_NONE))
+    if (aResetRadio)
     {
+        SuccessOrExit(SendReset(SPINEL_RESET_STACK));
+        IgnoreError(WaitResponse(false));
+        mError = OT_ERROR_NONE;
+
         otLogInfoPlat("Software reset RCP successfully");
         ExitNow(resetDone = true);
     }
 
     hardwareReset = (mSpinelInterface.HardwareReset() == OT_ERROR_NONE);
-    SuccessOrExit(WaitResponse(false));
```

Comment:
I think we should keep this.

`WaitResponse()` seems to also wait till `mIsReady` is set (which is set when we receive a `RESET` spinel event frame from RCP, indicating it was reset).

Reply:
@abtink The problem in here is that for uart ransmission the `HardwareReset` is not implemented https://github.com/openthread/openthread/blob/main/src/posix/platform/hdlc_interface.hpp#L148C13-L148C26
meaning it's not triggering anything. So `WaitResponse` is timing out.

Reply:
I see. Got it. Thanks.

How about adding `if (harwareReset)` check around `WaitResponse()` call? (so we only do it when we don't get an error?)

Reply:
I can tweak it this way.

Reply:
The `WaitResponse(false))` here is used for handling two case:
- If the spinel interface support hardware reset, the `WaitResponse(false))` is used for waiting for RESET reason response.
- If the `aResetRadio ` is set to false and the Spinel interface doesn't support the hardware reset.  The `WaitResponse(false))` is used for waiting for RESET reason. When the RCP starts, it sends RESET reason automatically to the host. The openthread simulation uses this case.

In your case, I assume that the  `aResetRadio ` is set to `true`. If failed to reset the RCP stack via the spinel commands, the `WaitResponse(false)` should time out and the host should die.

Reply:
In general, You are correct in the case when the `HardwareReset` is implemented.
In case of HDLC uart connection it's not so `WaitResponse` won't receive this status from the RCP.

Reply:
Looks good to me. Thanks.